src/bot/index.ts

```diff
@@ -1,25 +1,60 @@
 import { Probot } from 'probot'
 import { trpcServer } from '../server/trpc'
 import { logger } from '../utils/logger'
-import { branchProtectionGQL } from './graphql'
+import {
+  branchProtectionRulesetGQL,
+  getBranchProtectionRulesetGQL,
+} from './graphql'
 
 const botLogger = logger.getSubLogger({ name: 'bot' })
 
-// Helper function to get the metadata from the repository description
-export const getMetadata = (description: string | null) => {
-  botLogger.debug('Getting metadata from repository description', {
-    description,
-  })
+// Helper function to get the fork name from the repository custom properties
+export const getForkName = async (context: any) => {
+  return context.payload.repository.custom_properties.fork
+    ? context.payload.repository.custom_properties.fork
+    : null
+}
 
-  if (!description) {
-    return {}
-  }
+// Helper function to create branch protection rulesets
+export const createBranchProtectionRuleset = async (
+  context: any,
+  authenticatedApp: any,
+  ruleName: string,
+  includeRefs: string[],
+) => {
+  // Get the current branch protection rulesets
+  const getBranchProtectionRuleset = await context.octokit.graphql(
+    getBranchProtectionRulesetGQL,
+    {
+      owner: context.payload.repository.owner.login,
+      name: context.payload.repository.name,
+    },
+  )
+
+  if (
+    getBranchProtectionRuleset &&
+    getBranchProtectionRuleset.repository.rulesets.nodes.find(
+      (ruleset: any) => ruleset.name === 'default-branch-protection-icf',
+    )
+  ) {
+    botLogger.info('Branch protection rule already exists', {
+      getBranchProtectionRuleset,
+    })
+  } else {
```

Comment:
nit: prefer `return` as a control flow

Makes it a bit easier to read. So this would turn into

```
 if (
    getBranchProtectionRuleset &&
    getBranchProtectionRuleset.repository.rulesets.nodes.find(
      (ruleset: any) => ruleset.name === 'default-branch-protection-icf',
    )
  ) {
    botLogger.info('Branch protection rule already exists', {
      getBranchProtectionRuleset,
    })

   return;
}
```

Now the next statement isn't nested and makes it a little bit easier to follow